hermes-console-vue/src/views/topic/topic-form/TopicForm.vue

```diff
@@ -0,0 +1,315 @@
+<script setup lang="ts">
+  import { computed, ref } from 'vue';
+  import { useAppConfigStore } from '@/store/app-config/useAppConfigStore';
+  import { useCreateTopic } from '@/composables/topic/use-create-topic/useCreateTopic';
+  import { useEditTopic } from '@/composables/topic/use-edit-topic/useEditTopic';
+  import { useGlobalI18n } from '@/i18n';
+  import { useImportTopic } from '@/composables/topic/use-import-topic/useImportTopic';
+  import { useNotificationsStore } from '@/store/app-notifications/useAppNotifications';
+  import ConsoleAlert from '@/components/console-alert/ConsoleAlert.vue';
+  import SelectField from '@/components/select-field/SelectField.vue';
+  import TextField from '@/components/text-field/TextField.vue';
+  import type { TopicWithSchema } from '@/api/Topic';
+
+  const props = defineProps<{
+    topic: TopicWithSchema | null;
+    group: string | null | undefined;
```

Comment:
when can this be undefined?

Reply:
nice catch, it cannot be undefined, fixed